### PR TITLE
auto fix lint issues via CI

### PR DESCRIPTION
**Pull request by neo773**
This PR adds lint:fix job that can be triggered via `ci:lint` label.

/claim #505

### CODE REVIEW

File: .github/workflows/bot.yml [hunk 1]
```diff
@@ -113,3 +113,39 @@ jobs:
...
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - uses: actions/checkout@v4
+      - name: Run lint script
+        run: ./lint.sh
+      - name: Commit and push if changed
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: "style: apply lint fixes"
```
Comment (tusharmath) [inline .github/workflows/bot.yml:151]:
```suggestion
          commit_message: "style: apply lint fixes"  
```

**Comment by tusharmath** [comment]
@neo773 the CI doesn't seem to be fixing the lint issues. Can you have a look

**Comment by neo773** [comment]
@tusharmath 
Had a successful run here https://github.com/neo773/tailcall/pull/10
I think the PR needs to be merged first so it has the proper permissions.

**Comment by ologbonowiwi** [comment]
@tusharmath, you'll likely need to change the repo settings.

look at [this comment](https://github.com/JamesIves/github-pages-deploy-action/issues/1110#issuecomment-1117481234) on JamesIves/github-pages-deploy-action#1110; you'll need to give read and write on workflow permissions

File: .github/workflows/bot.yml [hunk 2]
```diff
@@ -113,3 +113,39 @@ jobs:
...
```
Comment (tusharmath) [inline .github/workflows/bot.yml:151]:
```suggestion
          commit_message: "style: apply lint fixes"
```

**Review by tusharmath — APPROVED** [review]

**Review by tusharmath — APPROVED** [review]

**Pull request merged**